### PR TITLE
task 8 completed

### PR DESCRIPTION
**Pull request by kajihan**

### CODE REVIEW

File: src/pro/java/hw8/ArrayValueCalculator.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            int calcArray = doCalc(array);
+            System.out.println("Summary result: " + calcArray);
+        } catch (ArraySizeException | ArrayDataException e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    public static int doCalc(String[][] array) throws ArraySizeException, ArrayDataException {
+        if (array.length != 4 || array[0].length != 4) {
+            throw new ArraySizeException("Incorrect array size");
```
Comment (kartius) [inline src/pro/java/hw8/ArrayValueCalculator.java]: What I recommend here  - is to overload constructor of ArrayDataException with additional parameter Trowable and create ArrayDataException with message and Exception from catch block do not loose original stacktrace of exception

Comment (kajihan) [reply src/pro/java/hw8/ArrayValueCalculator.java]: done

File: src/pro/java/hw8/ArrayValueCalculator.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+            for (int i = 0; i < array.length; i++) {
+                for (int j = 0; j < array[i].length; j++) {
+                    String valueStr = array[i][j];
+                    try {
+                        int value = Integer.parseInt(valueStr);
+                        result += value;
+                    } catch (NumberFormatException e) {
+                        try {
+                            throw new ArrayDataException("Invalid value at element [" + i + "][" + j + "]", e);
```
Comment (kartius) [inline src/pro/java/hw8/ArrayValueCalculator.java]: You don't need this block here:
                        try {
                            throw new ArrayDataException("Invalid value at element [" + i + "][" + j + "]", e);
                        } catch (Throwable throwable) {
                            throwable.printStackTrace();
                            return 0;
                        }

Will be enough just to have:
throw new ArrayDataException("Invalid value at element [" + i + "][" + j + "]", e);

File: src/pro/java/hw8/ArrayValueCalculator.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                        }
+                    }
+                }
+            }
+            return result;
+        } catch (Exception e) {
+            try {
+                throw new ArrayDataException("Error processing array", e).getCause();
+            } catch (Throwable throwable) {
+                throwable.printStackTrace();
```
Comment (kartius) [inline src/pro/java/hw8/ArrayValueCalculator.java]: The same here

File: src/pro/java/hw8/ArrayValueCalculator.java (hunk 2 above)
Comment (kajihan) [reply src/pro/java/hw8/ArrayValueCalculator.java]: fixed

File: src/pro/java/hw8/ArrayValueCalculator.java (hunk 3 above)
Comment (kajihan) [reply src/pro/java/hw8/ArrayValueCalculator.java]: fixed, but then we won't be able to see the stack trace if exception happens

**Review by kartius — APPROVED** [review]

**Pull request merged**